tests/test_validation.py

```diff
@@ -1352,6 +1342,43 @@ def test_rows(self):
         self.assertEqual(df['row2'][1], 3)
         self.assertEqual(df['row2'].dtype, float)
 
+    def test_recursive_path_validation(self):
```

Comment:
Should there also be a test case for vector paths?

Reply:
Sure! I added a couple more tests